common/main/cpp/native_c4collection.cc

```diff
@@ -318,29 +318,29 @@ Java_com_couchbase_lite_internal_core_impl_NativeC4Collection_createVectorIndex(
         jlong coll,
         jstring jName,
         jstring jqueryExpressions,
-        jint dimensions,
+        jlong dimensions,
+        jint metric,
         jlong centroids,
         jint encoding,
-        jint bits,
-        jint subquantizers,
-        jint metric,
+        jlong subquantizers,
+        jlong bits,
         jlong minTrainingSize,
         jlong maxTrainingSize) {
     C4IndexOptions options = {};
-    options.vector = {};
 
-    options.vector.encoding = kC4VectorEncodingSQ8; // !!! fix me
-    // options.vector.dimensions = dimensions;
-    // options.vector.encoding = { encoding, subquantizers, bits }
-
-    options.vector.numCentroids = centroids;
-    options.vector.metric = (C4VectorMetric) metric;
-    options.vector.minTrainingSize = minTrainingSize;
-    options.vector.maxTrainingSize = maxTrainingSize;
+    options.vector = {};
+    options.vector.dimensions = (unsigned) dimensions;
+    options.vector.metric = (C4VectorMetricType) metric;
+    options.vector.clustering = {kC4VectorClusteringFlat, (unsigned) centroids, 0, 0};
+    options.vector.encoding = { (C4VectorEncodingType) encoding, (unsigned) subquantizers, (unsigned) bits };
```

Comment:
FYI : C4VectorEncodingType has the first item in the enum as kC4VectorEncodingDefault so need to make sure that the encoding value passed from Java has offset that.

Reply:
A weird choice when it could just be aliased to another value in LiteCore but good to know...